### DISCUSSION TITLE
[Bug] Calling reset with values and keepDefaultValues doesn't update values

### DISCUSSION THREAD

**Discussion by mkrnkv** [react-hook-form/react-hook-form#6710] [Q&A] [answered]
Hey,

I'm using react-hook-form with material ui components via useController. I'm attempting to reset a form with new values while keeping the defaultValues unchanged. I see with getValues that immediately after calling `reset(values, { keepDefaultValues: true } `, the values are changed, but this is not reflected in the input components and after this the values are reset back to defaultValues.

Am I misunderstanding the use for reset or is there some error in my logic? I wouldn't want to use setValue as I'll need to update many fields at the same time.

Here's a codesandbox link demonstrating the issue: https://codesandbox.io/s/upbeat-robinson-zph9n

By looking at console logs we can see that after pressing 'Load values' the form values are changed.

**Comment by mkrnkv** [comment, marked as answer]
Seems like `reset(values, { keepDefaultValues: true }` works in my example in versions before 7.9.0.

Here's a simplified CSB that shows the issue: https://codesandbox.io/s/nameless-monad-fvkt2

If you go back to e.g. version 7.8.8 the reset will actually change the input values.

**Reply by bluebill1049** [reply]
thanks for the details above.

`reset({ name: "loaded" }, { keepDefaultValues: true });` this is actually conflicting, because if you want to keep default Values then the original defaultValues will be used to populate the input, just like defaultValues was supplied to useForm to populate your input, i think you want to use setValue instead with your use case, that way defaultValues will stay the same but only value gets updated.

**Reply by mkrnkv** [reply]
Thanks for the reply. So what's the use case for calling `reset(values, { keepDefaultValues: true })`, or is there none? I would prefer not to use setValue as my form will consist of tens of fields that need to be changed simultaneously, while also needing to keep the original defaultValues. Is there another way to achieve this? Logically `reset(values, { keepDefaultValues: true })` would make sense to me.

**Reply by bluebill1049** [reply]
you made a good point there, perhaps the library should invoke setValue internally on the user's behavior.

**Reply by bluebill1049** [reply]
could you submit a feature request and I will take a look at it tmr?

**Reply by mkrnkv** [reply]
Sure done #6721. Thanks!

**Comment by yashsway** [comment]
can confirm this doesn't work as you'd expect.
